File: apple/Harness/Services/FeatureFlagService.swift
//
//  Copyright 2022 Harness
//

import Foundation

protocol FeatureFlagService {
    func load(_ callback: @escaping()->())
    func destroy()
    func boolVariation(evaluationId: String, _ callback: @escaping(_ result: Bool)->())
    func stringVariation(evaluationId: String, _ callback: @escaping(_ result: String)->())
    func numberVariation(evaluationId: String, _ callback: @escaping(_ result: Int)->())
    func jsonVariation(evaluationId: String, _ callback: @escaping(_ result: String)->())
}

File: android/app/src/main/java/io/harness/services/FeatureFlagService.kt
package io.harness.services

typealias JSONString = String
typealias FeatureFlagCallback<T> = (T)->Unit

interface FeatureFlagService {

    fun load(callback: ()->Unit)
    fun destroy(callback: ()->Unit = {})

    fun boolVariation(evaluationId: String, callback: FeatureFlagCallback<Boolean>)
    fun stringVariation(evaluationId: String, callback: FeatureFlagCallback<String>)
    fun numberVariation(evaluationId: String, callback: FeatureFlagCallback<Number>)
    fun jsonVariation(evaluationId: String, callback: FeatureFlagCallback<JSONString>)

}